tephra-core/src/main/java/co/cask/tephra/TransactionManager.java

```diff
@@ -325,6 +328,10 @@ private void cleanupTimedOutTransactions() {
             // timed out, remember tx id (can't remove while iterating over entries)
             timedOut.add(tx.getKey());
             LOG.info("Tx invalid list: added tx {} because of timeout", tx.getKey());
+          } else if (expiration < 0) {
+            LOG.warn("Transaction {} has negative expiration time {}. " +
```

Comment:
Would it be better in this case to always expire the transaction?  Having a transaction sit around forever doesn't seem good.

If it's actually due to a problem in migration, we might actually want to log a fatal error and abort the service so that we are not running with invalid data.  Though the problem with that is we need a way of clearing the state so startup can succeed.  So based on that I think it might be better to just expire the transaction.